helix-core/src/main/java/org/apache/helix/controller/stages/BestPossibleStateCalcStage.java

```diff
@@ -90,60 +96,86 @@ private BestPossibleStateOutput compute(ClusterEvent event, Map<String, Resource
 
     BestPossibleStateOutput output = new BestPossibleStateOutput();
 
-    for (String resourceName : resourceMap.keySet()) {
-      logger.debug("Processing resource:" + resourceName);
+    // Reset current INIT/RUNNING tasks on participants for throttling
+    JobRebalancer.resetActiveTaskCount(cache.getLiveInstances().keySet(), currentStateOutput);
 
-      Resource resource = resourceMap.get(resourceName);
-      // Ideal state may be gone. In that case we need to get the state model name
-      // from the current state
-      IdealState idealState = cache.getIdealState(resourceName);
+    PriorityQueue<JobResourcePriority> jobResourceQueue = new PriorityQueue<JobResourcePriority>();
```

Comment:
Got your point.
But we don't need to use all features of the queue to leverage it here.
If the performance is bad, we shall consider an alternative. But that's not the case.
And even we use "sortedSet", there is no usage of a "set" here. So, to me, there is no different.